openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/LineageRepository.java

```diff
@@ -90,14 +90,13 @@ private String validateLineageDetails(EntityReference from, EntityReference to,
     }
 
     List<ColumnLineage> columnsLineage = details.getColumnsLineage();
-    if (areValidEntities(from, to)) {
-      throw new IllegalArgumentException(
-          "Column level lineage is only allowed between two tables or from table to dashboard.");
-    }
-
-    Table fromTable = dao.tableDAO().findEntityById(from.getId());
-    ColumnsEntityInterface toTable = getToEntity(to);
-    if (columnsLineage != null) {
+    if (columnsLineage != null && !columnsLineage.isEmpty()) {
+      if (areValidEntities(from, to)) {
+        throw new IllegalArgumentException(
+            "Column level lineage is only allowed between two tables or from table to dashboard.");
```

Comment:
we don't need to check this. You can have lineage between table to chart as well, topic to table, pipeline to table etc..

Reply:
@harshach this is for the column level lineage.